packages/payload/src/auth/operations/forgotPassword.ts

```diff
@@ -143,7 +143,7 @@ export const forgotPasswordOperation = async <TSlug extends CollectionSlug>(
       Date.now() + (collectionConfig.auth?.forgotPassword?.expiration ?? expiration ?? 3600000),
     ).toISOString()
 
-    user = await payload.update({
+    user = await payload.db.updateOne({
```

Comment:
Reverse this change and let's change the structure of the user update data instead so this and the lines above it look like the following:

```ts
const resetPasswordExpiration = new Date(
  Date.now() + (collectionConfig.auth?.forgotPassword?.expiration ?? expiration ?? 3600000),
).toISOString()

user = await payload.update({
  id: user.id,
  collection: collectionConfig.slug,
  data: {
    resetPasswordExpiration,
    resetPasswordToken: token,
  },
  req,
})
```

You can safely remove the line re-assigning the `user.resetPasswordToken` value entirely as well